### PR TITLE
Model evaluation

### PR DESCRIPTION
**Pull request by zahraaghapour**
Please review the evaluation module.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -44,4 +45,11 @@
...
     # train the model   
-    model.fit(train_X, train_y, epochs=args.n, batch_size=32, validation_data=(test_X, test_y), callbacks=Callbacks)
+    History = model.fit(train_X, train_y, epochs=args.n, batch_size=32, validation_data=(test_X, test_y), callbacks=Callbacks)
+    
+    # Model evaluation
+    # plot train/validation loss/accuracy figures
+    plot_history(History, epoch=args.n, metric='accuracy')
+    plot_history(History, epoch=args.n, metric='loss')
+    # print metrics
+    print_evaluation_metrics(model, test_X, test_y)
```
Comment (parisa-ahmadi) [inline train.py:55]: Train section sounds good..

File: evaluation.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      predictions = model.predict(test_X, batch_size=32)
+      # Convert probability vector into target vector
+      predictions= np.where(predictions > 0.5, 1, 0)
+      predictions.reshape(predictions.shape[0],)
+      # print the metrics
+      print("\nConfusion matrix: \n\n",metrics.confusion_matrix(test_Y, predictions))
+      print('\n       -----------------     ')
+      print('\n',classification_report(test_Y, predictions, target_names=['non smile','smile']))
+      print('\n       -----------------     ')
+      print("\nAUC: ",metrics.roc_auc_score(test_Y, predictions))
```
Comment (parisa-ahmadi) [inline evaluation.py:30]: Suggestion:
You can add Detection part with take a image and detect and plot it finally ,The code would be great with this section..
like this:

def predict_class(path):
    img = cv2.imread(path)

    RGBImg = cv2.cvtColor(img,cv2.COLOR_BGR2RGB)
    RGBImg= cv2.resize(RGBImg,(224,224))
    plt.imshow(RGBImg)
    image = np.array(RGBImg) / 255.0
    new_model = tf.keras.models.load_model("64x3-CNN.model")
    predict=new_model.predict(np.array([image]))
    per=np.argmax(predict,axis=1)
    if per==1:
        print('No Smile')
    else:
        print('Smile')

predict_class('The address of that image to detect its class')

**Review by parisa-ahmadi — CHANGES_REQUESTED** [review]
Suggestion:
You can add Detection part with take a image and detect and plot it finally ,The code would be great with this section..
like this:

def predict_class(path):
    img = cv2.imread(path)

    RGBImg = cv2.cvtColor(img,cv2.COLOR_BGR2RGB)
    RGBImg= cv2.resize(RGBImg,(224,224))
    plt.imshow(RGBImg)
    image = np.array(RGBImg) / 255.0
    new_model = tf.keras.models.load_model("64x3-CNN.model")
    predict=new_model.predict(np.array([image]))
    per=np.argmax(predict,axis=1)
    if per==1:
        print('No Smile')
    else:
        print('Smile')

predict_class('The address of that image to detect its class')

**Review by parisa-ahmadi — CHANGES_REQUESTED** [review]
Suggestion:
You can add Detection part with take a image and detect and plot it finally ,The code would be great with this section..
like this:

def predict_class(path):

    img = cv2.imread(path)
    RGBImg = cv2.cvtColor(img,cv2.COLOR_BGR2RGB)
    RGBImg= cv2.resize(RGBImg,(224,224))
    plt.imshow(RGBImg)
    image = np.array(RGBImg) / 255.0
    new_model = tf.keras.models.load_model("64x3-CNN.model")
    predict=new_model.predict(np.array([image]))
    per=np.argmax(predict,axis=1)
    if per==1:
        print('No Smile')
    else:
        print('Smile')

predict_class('The address of that image to detect its class')
